.github/workflows/kit-bootstrap.yml

```diff
@@ -0,0 +1,135 @@
+          elif [[ "${{ matrix.os }}" == "macos-latest" ]]; then
+            curl -LO "https://storage.googleapis.com/kubernetes-release/release/$(curl -L -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/darwin/amd64/kubectl"  
+          fi
+          chmod +x kubectl
+          sudo mv kubectl /usr/local/bin/
+      - name: Builds the KIT CLI ,Provisions the Kit Environment,KIT guest cluster and deletes.
+        run: |
+          cd ${HOME}/work/kubernetes-iteration-toolkit/kubernetes-iteration-toolkit/substrate/cmd/kitctl
+          go build  .
+          if [[ "${{ matrix.os }}" == "ubuntu-latest" ]]; then
+            os_type="linux"    
+          elif [[ "${{ matrix.os }}" == "macos-latest" ]]; then
+            os_type="darwin"
+          fi 
+          ./kitctl bootstrap kit-github-${os_type}-${{ github.run_id }}-${{github.run_attempt}}
+          export KUBECONFIG=${HOME}/.kit/env/kit-github-${os_type}-${{ github.run_id }}-${{github.run_attempt}}/etc/kubernetes/admin.conf
+          kubectl get pods -A
+          for i in 21 22 23 24 25; do
+            GUEST_CLUSTER_NAME="guest-${os_type}-1-${i}-${{ github.run_id }}-${{github.run_attempt}}"
+            K8S_VERSION=$(echo ${GUEST_CLUSTER_NAME} | sed 's/guest-linux-1-//;s/-.*//;s/^/1./')
+            cat <<-EOF | kubectl apply -f -
+              apiVersion: kit.k8s.sh/v1alpha1
+              kind: ControlPlane
+              metadata:
+                name: ${GUEST_CLUSTER_NAME} # Desired Cluster name
+              spec:
+                kubernetesVersion: "${K8S_VERSION}"
+                etcd:
+                  replicas: 3
+                master:
+                  apiServer:
+                    replicas: 1  
+          EOF
+           for (( time=0; time<=1000; time+=5 )); do
+              controlplane_object=$(kubectl get controlplane ${GUEST_CLUSTER_NAME}  -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}' 2>/dev/null)
+              if [[ "$controlplane_object" == "True" ]]; then
+                echo "Control plane objects are ready"
+                break;
+              else
+                echo "Control plane objects are not ready"
+              fi
+              if [[ "$time" -eq 990 ]]; then
+                echo "Timeout: Control plane objects are not ready after 20 minutes"
+                kubectl logs deploy/kit-controller -n kit
+                exit 1
+              fi
+              sleep 5
+            done
+            kubectl wait --for=condition=Ready pods  --all --timeout=5m
+            kubectl get secret ${GUEST_CLUSTER_NAME}-kube-admin-config -ojsonpath='{.data.config}' | base64 -d > /tmp/kubeconfig
+            kubectl --kubeconfig=/tmp/kubeconfig apply -f https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/release-1.10/config/master/aws-k8s-cni.yaml
+            kubectl --kubeconfig=/tmp/kubeconfig rollout status daemonset/aws-node -n kube-system
```

Comment:
This is not doing anything useful when there are no worker nodes.
I'd suggest to change it to:
- Create DP object
- Check DP reporting ready in staus
- Apply CNI
- Use `rollout status` to wait for completion.